packages/react-native/ReactCommon/react/renderer/components/view/ViewShadowNode.cpp

```diff
@@ -72,6 +72,11 @@ void ViewShadowNode::initialize() noexcept {
       !viewProps.backgroundImage.empty() ||
       HostPlatformViewTraitsInitializer::formsView(viewProps) ||
       viewProps.outlineWidth > 0;
+    
+    if (viewProps.yogaStyle.display() == yoga::Display::Contents) {
+        formsStackingContext = false;
+        formsView = false;
+    }
```

Comment:
I think ChildrenFormStackingContext could still cause the view to be materialized here. Also think this logic may not run for non-view-derived but still container components. Think we might need to expose this to Differentiator in different way.

Reply:
I think you're right. I've moved the logic to Differentiator and added a new trait that allows a view to be force-flattened in https://github.com/facebook/react-native/pull/46584/commits/624969b0712086b8ce5cec796a4fb2b0d880a5d2. What do you think about this approach?